### DISCUSSION TITLE
Dimensions flipped based on image orientation when cropping

### DISCUSSION THREAD

**Discussion by neolution-ch2** [SixLabors/ImageSharp#1339] [Q&A]
I tried to create a gallery with pictures of different orientations. Mostly they are landscape but some of them are portraits. For the thumbnails I specified `width: 240` , `height: 160` and `rmode: crop`. 

And it worked for all landscape-oriented pictures. But for the ones in portrait orientation it flipped the specs upside down: it cropped the thumbnail with 160x240 (instead of 240x160).

I don't think it's a bug and intentional. But can someone explain to me what and why this is happening? And is there a way to avoid this behavior?

**Comment by JimBobSquarePants** [comment]
It sounds like the images are actually landscape but have the [EXIF orientation mode](https://www.impulseadventure.com/photo/exif-orientation.html) set to rotate the images to portrait.

Using the same dimensions across all should work or else use `AutoOrient()` to fix the orientation before resizing.

**Reply by toy-opensource** [reply]
autoorient was key  - thx!

**Comment by neolution-ch2** [comment]
Checked it in a EXIF viewer, and you're right. I didn't know that.

![image](https://user-images.githubusercontent.com/5662575/92211675-19dffe00-ee91-11ea-8839-101faffb1c12.png)

What do you mean "Using the same dimensions across all should work"? I use the same params for all pictures.

Thanks for the `AutoOrient()` method, I may have to use it... But I really didn't plan to touch user-uploaded pictures,

**Reply by tocsoft** [reply]
This makes me think we might want an API explicitly to return the 'Display' dimensions rather than to storage dimensions (which is what we show in ImageSharp). Just a little extension method that can take takes the metadata into account and report how the visual dimensions. (API would need the caveat that it will only apply for user agents that respect orientation and resolution etc). Might even want it to be able (as additional property) to return physical size based on resolution as well.

**Reply by toy-opensource** [reply]
yes @neolution-ch - i wasnt planning on touching the photos either
